CMakeLists.txt

```diff
@@ -0,0 +1,223 @@
+cmake_minimum_required (VERSION 3.12)
+project(Marabou)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+option(BUILD_PYTHON "Build Python" ON)
+option(FORCE_PYTHON_BUILD "Build python even if there is only python32" OFF)
+option(RUN_UNIT_TEST "run cxxtest unit testing" OFF)
+option(RUN_MEMORY_TEST "run cxxtest unit testing with ASAN" ON)
+option(RUN_REGRESS_TEST "run regression tests" OFF)
+
+set(MARABOU_LIB MarabouHelper)
+set(MARABOU_TEST_LIB MarabouHelperTest)
+set(MARABOU_EXE Marabou)
+set(MARABOU_REGRESS Regression)
+set(MARABOU_PY MarabouCore)
+
+set(TOOLS_DIR "${PROJECT_SOURCE_DIR}/tools")
+set(SRC_DIR "${PROJECT_SOURCE_DIR}/src")
+set(PYTHON_API_DIR "${PROJECT_SOURCE_DIR}/maraboupy")
+set(REGRESS_DIR "${PROJECT_SOURCE_DIR}/regress")
+
+set(ENGINE_DIR "${SRC_DIR}/engine")
+set(PYBIND11_DIR "${TOOLS_DIR}/pybind11-2.3.0")
+set(BOOST_DIR "${TOOLS_DIR}/boost_1_68_0")
+set(COMMON_DIR "${SRC_DIR}/common")
+set(BASIS_DIR "${SRC_DIR}/basis_factorization")
+set(ACAS_REGRESS_NET "${REGRESS_DIR}/acas_nnet")
+
+
+set(COMMON_REAL "${COMMON_DIR}/real")
+set(COMMON_MOCK "${COMMON_DIR}/mock")
+file(GLOB SRCS_COMMON_REAL "${COMMON_REAL}/*.cpp")
+file(GLOB SRCS_COMMON_MOCK "${COMMON_MOCK}/*.cpp")
+
+# We build a static library that is the core of the project, the link it to the
+# API's (executable and python at the moment)
+add_library(${MARABOU_LIB} ${SRCS_COMMON_REAL} )
+add_executable(${MARABOU_EXE} ${ENGINE_DIR}/main.cpp)
+add_executable(${MARABOU_REGRESS} ${REGRESS_DIR}/main_regress.cpp)
+
+
+# Build regress 
+target_link_libraries(${MARABOU_REGRESS} PRIVATE ${MARABOU_LIB})
+
+target_include_directories(${MARABOU_REGRESS} PRIVATE ${LIBS_INCLUDES})
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/logs)
+file(COPY ${ACAS_REGRESS_NET} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}) 
+file(COPY "${REGRESS_DIR}/lp_feasible_1.mps" DESTINATION ${CMAKE_CURRENT_BINARY_DIR}) 
```

Comment:
Is it possible to move this logic (listing of specific files) to the CMakeLists under regress?